doc/manual/src/contributing/testing.md

```diff
@@ -2,10 +2,57 @@
 
 ## Unit-tests
 
-The unit-tests for each Nix library (`libexpr`, `libstore`, etc..) are defined
-under `src/{library_name}/tests` using the
-[googletest](https://google.github.io/googletest/) and
-[rapidcheck](https://github.com/emil-e/rapidcheck) frameworks.
+The unit-tests for defined using the [googletest] and [rapidcheck] frameworks.
+
+[googletest]: https://google.github.io/googletest/
+[rapidcheck]: https://github.com/emil-e/rapidcheck
+
+### Source and header layout
+
+> An example for `libnixexpr`, demonstrating much of what is described below
+>
+> ```
+> src
+> ├── libexpr
+> │   ├── value/context.hh
+> │   ├── value/context.cc
+> │   │
+> │   …
+> ├── libexpr-tests
+> │   ├── tests/
+> │   │   ├── value/context.hh
+> │   │   │
+> │   │   …
+> │   ├── value/context.cc
+> │   │
+> │   …
+> …
+> ```
+
+The tests for each Nix library (`libnixexpr`, `libnixstore`, etc..) live inside a directory `src/${library_name_without-nix}-tests` next to the directory for the library (`src/${library_name_without-nix}`).
```

Comment:
Scrolling back and forth I understand what you mean by "without-nix", but would it hurt so much to not have that distinction? E.g. by removing the nix in the original library name or adding it for the tests as well? It makes something that's fairly involved and obscure needlessly hard to grok.

Reply:
that was actually discussed as a future goal of current refactoring efforts :) the problem currently is striking a balance between discrete specific small improvements that can be introduced overtime, and a just one huge commit that changes everything. seeing as the current layout uses libnix* to discuss nix libraries, it would be best to keep as is for now imo

Reply:
Yeah the other dir is `libexpr` so it seems prudent to let that be for now and call this `libexpr-tests`